content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,289 @@
+---
+layout: blog
+title: 'Kubernetes v1.31: TITLE'
+date: 2024-08-14
+slug: kubernetes-v1-31-release
+author: >
+  [Kubernetes Release Team](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.31/release-team.md)
+---
+**Authors:** [Kubernetes 1.31 Release Team](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.31/release_team.md)
+
+
+
+## Release theme and logo
+<Logo image size is recommended to be no more than 2160px/>
+
+## Highlights of features graduating to Stable
+
+_This is a selection of some of the improvements that are now stable following the v1.31 release._
+
+### Improved ingress connectivity reliability for Kube-proxy
+
+Kube-proxy improved ingress connectivity reliability is GA in v1.31. One of the common problems with Load Balancers in Kubernetes is the synchronization between the different components involved to avoid traffic drop, this feature implements a mechanism in kube-proxy for load balancers to do connection draining for terminating Nodes and establish some best practices for cloud providers and Kubernetes load balancers implementations.
+
+
+exerpt from SIG:
+This feature enables connection draining for workloads running on terminating/deleting Kubernetes nodes which are exposed by services of type: LoadBalancer and externalTrafficPolicy: Cluster . Note: the predicates for using this feature are: a) kube-proxy needs to run as default service proxy on the cluster and b) the load balancer needs to support connection draining. There are no specific changes required for using this feature, it has been enabled by default in Kube-proxy since 1.30 and been promoted to stable in 1.31.
+
+This work was done as part of [KEP #3836](https://github.com/kubernetes/enhancements/issues/3836) by [SIG Network](https://github.com/kubernetes/community/tree/master/sig-network). /docs/reference/networking/virtual-ips/#external-traffic-policy
+
+### AppAprmor support is now stable
+
+Kubernetes support for AppArmor is now GA. Protect your Containers using AppArmor by setting the `appArmorProfile.type` field in the Container's `securityContext`. Note that before Kubernetes v1.30, AppArmor was controlled via annotations; starting in v1.30 it is contorlled using fields. Its recommended that you should migrate away from using annotations and start using the `appArmorProfile.type` field.
+
+To learn more read the [AppArmor tutorial](/docs/tutorials/security/apparmor/). This work was done as a part of [KEP #24](https://github.com/kubernetes/enhancements/issues/24) lead by  [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
```

Comment:
Do we have any other docs for AppArmor yet? Ideally we would.

Reply:
as far as i can tell these are the docs we have. There is a [PR](https://github.com/kubernetes/website/pull/47254) for adding some more apparmor related docs, but its not merged yet.

Reply:
@sftim following up here,  dont think there are any other apparmor docs that i can find.

Reply:
We could nudge release docs to nudge the SIG about adding some, but it's not directly relevant to this announcement.